### PR TITLE
Defect/2 fa refactor

### PR DESCRIPTION
**Pull request by fareedmohammed**

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -212,8 +213,10 @@ module.exports = function createApp({ signInService }, logger, calendarService,
     })(req, res, next),
   )
 
-  app.use('/logout', (req, res) => {
+  app.use('/logout', async (req, res) => {
     if (req.user) {
+      await userAuthenticationService.updateSessionExpiryDateTime(req.user.username)
```
Comment (petergphillips) [inline server/app.js:218]: not following this change - why do you need to modify track the session expiry time?  Isn't this just the expiry time of the cookie?

File: server/app.js [hunk 2]
```diff
@@ -223,8 +226,8 @@ module.exports = function createApp({ signInService }, logger, calendarService,
 
   // Routing
   app.use('/', standardRoute(createLoginRouter()))
-  app.use('/calendar', authHandler, standardRoute(createCalendarRouter(logger, calendarService, notificationService)))
-  app.use('/details', authHandler, standardRoute(createCalendarDetailRouter(logger, calendarService)))
+  app.use('/calendar', authHandler, standardRoute(createCalendarRouter(logger, calendarService, notificationService, userAuthenticationService)))
+  app.use('/details', authHandler, standardRoute(createCalendarDetailRouter(logger, calendarService, userAuthenticationService)))
   app.use('/notifications', authHandler, standardRoute(createNotificationRouter(logger, notificationService)))
```
Comment (petergphillips) [inline server/app.js:231]: why does `/notifications` not need the user authentication code too?

Comment (petergphillips) [reply server/app.js:231]: ah because you don't need the notification api url from it

Comment (fareedmohammed) [reply server/app.js:231]: Yes it does, this is missed by me. I'll fix.

File: server/app.js (hunk 1 above)
Comment (fareedmohammed) [reply server/app.js:218]: Where is the expiry time of cookie set?

**Pull request merged**